File: src/combat/DamageInstance.cpp
/**
 * This is the damage instance class. It will be the handler of damage
 */
#include "Chara.hpp"


DamageInstance::DamageInstance() {
  Chara;
}

//Physical attack done on to character
int DamageInstance::physAtk(int d, Chara c) {
  int hp = c.hp;
  int full_damage = c.action1;
  int reduced_damage = full_damage - c.block;
  return hp-reduced_damage;
}

//Fire attack done on to character
int DamageInstance::fireAtk(int d, Chara c) {
  int hp = c.hp;
  int full_damage = c.action1;
  int reduced_damage = full_damage - c.fres;
  return hp-reduced_damage;
}

//Cold attack done on to character
int DamageInstance::coldAtk(int d, Chara c) {
  int hp = c.hp;
  int full_damage = c.action1;
  int reduced_damage = full_damage - c.cres;
  return hp-reduced_damage;
}

//Light attack done on to character
int DamageInstance::ligtAtk(int d, Chara c) {
  int hp = c.hp;
  int full_damage = c.action1;
  int reduced_damage = full_damage - c.lres;
  return hp-reduced_damage;
}

File: lua/DamageInstance.lua
-------Damage Instance-------

--Damage instance will take in an ability with attributes, and a character
--then will inflict the character with the abilities effects... poison, spell_damage, 

--for now... send in the damage and type that will be inflicted onto the character
local DamageInstance =  {}

function DamageInstance.Damage(damageT, character_name)
	local character = require(character_name)
	--inflict damage on character C
	local attack_dmg = damageT[1]
	local attack_type = damageT[2]

	local reduced_damage = attack_dmg - resistance(attack_type,character)

	character.hp = character.hp - reduced_damage

end
	


function resistance(type,character)
	local resist = -1
	if type == 'physical' then
		resist = character.armour
	elseif type == 'fire' then
		resist = character.fres
	elseif type == 'cold' then
		resist = character.cres
	elseif type == 'light' then
		resist = character.lres
	end

	assert(resist ~= -1)
	return resist
end

return DamageInstance


